### ISSUE TITLE
DOC update fairlearn repository overview

### ISSUE THREAD

**Issue by hildeweerts** [fairlearn/fairlearn#900]
#### Describe the issue linked to the documentation

The [Fairlearn repository overview](https://fairlearn.org/main/contributor_guide/fairlearn_repository_overview.html) on our website is outdated. In particular, Visualization source code is no longer in the visualization directory.

#### Suggest a potential alternative/fix

* remove "Visualization source code in the visualization directory"
* add description of scripts directory
* add description of notebooks directory
* add description of test_othermlpackages directory

**Comment by romanlutz** [comment]
Taking a step back: is that a useful page to have? [I created it so I don't feel bad questioning this 🤣 ]

**Comment by hildeweerts** [comment]
I can imagine that the repo can be a bit intimidating for new folks looking to contribute to the project, although right now there might not be enough information on this page for it to be really useful.

I'd be curious to hear what some of our newer contributors think @shimst3r @bram49 @alexquach @kurianbenoy

**Comment by shimst3r** [comment]
I stumbled upon that page by chance, but didn't get anything out of it very much (sorry @romanlutz :sweat_smile:).

What *might* be a good idea in my opinion would to move this section out of the documentation, into the README (or somewhere else closer to the "raw" source code). That's where I'd look for an explanation of the repository intuitively and have so for other projects in the past.

This could be used to explain files like `requirements-customplots.txt` as well, or why there is a `pyproject.toml`, `setup.py`, **and** `setup.cfg`.

That's something I'd expect from a well-maintained project like Fairlearn! :nerd_face:

**Comment by adrinjalali** [comment]
My preference would be somewhere in the documentation, linked from the contribution guidelines. But I don't have a strong preference as long as people can find it while reading the contributing guidelines.

**Comment by ojeda-e** [comment]
Hi, I would like to help with a good first issue. Is this still an option for new contributors @hildeweerts? 
The layout I have in mind to start looks like the following:

Title
[Same short introductory paragraph as in [README](https://github.com/fairlearn/fairlearn#fairlearn)]

- Tests
[Add a short description of what tests are intended for. Unit tests, notebooks, etc]

- Visualization (delete)

- Automation
[Add a short description, goal]

- Documentation
   - API
    [Short description]
    - User guide
    [Short description]

- Examples
[List of examples with links]. We could potentially mention here that the API also includes examples when issue #894 is solved.

Thanks.

**Comment by adrinjalali** [comment]
I'm now wondering if this is a good first issue actually, since it requires a deep understanding of how the repo is structured. Hmm.

To me, the documentation should allow first time contributors and people who are not familiar with the repo to find their way around when they come here, and to know where to go for each type of contribution they'd like to have.

**Comment by hildeweerts** [comment]
For a first non-outdated version I'd be very happy with the layout suggested by @ojeda-e. WDYT @adrinjalali?

**Comment by adrinjalali** [comment]
I would change it a little bit.

It doesn't include the modules and the code.

For examples, I don't think we need to have a list of examples here, it'd be very quickly outdated, it makes more sense to tell people how they can find the right example file if they want to contribute there.

And for the CI, it needs to explain where the scripts are and which CIs we use and for what.

**Comment by hildeweerts** [comment]
Ah yes, I agree.

@ojeda-e do you feel comfortable adding these suggestions? As indicated by @adrinjalali this might not be such a good first issue after all, so if you don't feel like digging through the repo please feel free to drop this issue!

**Comment by ojeda-e** [comment]
Thanks for your answers @hildeweerts @adrinjalali.
If you don't mind I would it give a try with the suggestions from @adrinjalali. If it's not good enough, you may not have an issue fixed, but it'll be still useful to me. 
If you think there is a better issue to help with, I would appreciate your suggestions as well.

**Comment by hildeweerts** [comment]
Awesome, @ojeda-e! Let's give it a try then :)

**Comment by alliesaizan** [comment]
Hi @hildeweerts and @adrinjalali, I am going through the backlog and seeing this issue was never addressed. Would we still want to have a repository page? If so, could we close this issue and open a new one with the desired template so it's easy for a contributor to pick up?

**Comment by hildeweerts** [comment]
Hi @alliesaizan, I think a repository-like page could be useful for newcomers (if accurate), so I would be happy with the template suggested by @ojeda-e. But perhaps some of the contributors who joined more recently can chime in with their thoughts on whether this is (or could be) useful and, if so, what information should be in there.

Tagging @taharallouche @TamaraAtanasoska @StefanieSenger @Mariam-ke

**Comment by TamaraAtanasoska** [comment]
I have to admit that before seeing this issue I didn't know there was a page. I don't think I've missed it personally, and I am not sure if I'd vote for having one, more towards not needing it. The project has a pretty standard structure, and it won't be mysterious to most folks coming to contribute with at least some engineering experience. If we were to grow much more and have very opinionated repo structure opinions that aren't obvious immediately, I would lean more towards a yes.

**Comment by alliesaizan** [comment]
Yeah honestly I did not notice it either until I saw this issue. But I will wait to hear from others as well.

**Comment by hildeweerts** [comment]
Allright, I'd be pretty happy with removing this page then. I will update the issue to reflect this!
